File: app/src/main/java/com/shuoye/video/utils/network/response/Response.java
package com.shuoye.video.utils.network.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

/**
 * 返回结果封装
 *
 **/
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Accessors(chain = true)
public class Response<T> {
    private int code; // 返回的code
    private T data; // 具体的数据结果
    private String msg; // message 可用来返回接口的说明
}

File: app/src/main/java/com/shuoye/video/api/Response.kt
package com.shuoye.video.api

/**
 * 响应体
 * @program Video
 * @ClassName Response
 * @author shuoye
 * @create 2021-10-23 15:48
 **/
data class Response<T>(
    val code: Int,
    val data: List<T> = emptyList(),
    val msg: String,
    val page: Int
)
